src/api/Nhs.Appointments.Api/Notifications/UserRolesChangedNotifier.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.Extensions.Options;
+using Nhs.Appointments.Core;
+using Notify.Interfaces;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Nhs.Appointments.Api.Notifications;
+
+// Depends on an email template hosted in Gov Notify. See template.txt for details 
+public class UserRolesChangedNotifier(IAsyncNotificationClient notificationClient, IRolesStore rolesStore, ISiteSearchService siteService, IOptions<UserRolesChangedNotifier.Options> options) : IUserRolesChangedNotifier
+{
+    public async Task Notify(string user, string siteId, string[] rolesAdded, string[] rolesRemoved)
+    {
+        var roles = await rolesStore.GetRoles();
+        var rolesAddedNames = GetRoleNames(roles, rolesAdded);
+        var rolesRemovedNames = GetRoleNames(roles, rolesRemoved);
+
+        if (rolesAddedNames.Length == 0 && rolesRemovedNames.Length == 0)
+        {
+            // avoid sending pointless notifications when nothing has actually changed:
```

Comment:
Could we avoid the need for a comment by reversing the logic and nesting the main function logic, and having a private method HasRoleChanges so that the code reads appropriately without the need for a comment. I prefer to avoid comments 

e.g.
var hasRoleChanges = rolesAdded.Any() || rolesRemoved.Any()
if(hasRoleChanges) { ... do the sending }